lib/metasploit/framework/data_service/remote/http/core.rb

```diff
@@ -130,20 +142,17 @@ def make_request(request_type, path, data_hash = nil, query = nil)
 
       case response
         when Net::HTTPOK
-          # puts 'request sent successfully'
           return SuccessResponse.new(response)
         else
-          puts "HTTP #{request_type} request: #{uri.request_uri} failed with code: #{response.code} message: #{response.body}"
+          ilog "HTTP #{request_type} request: #{uri.request_uri} failed with code: #{response.code} message: #{response.body}"
           return FailedResponse.new(response)
       end
     rescue EOFError => e
-      puts "ERROR: No data was returned from the server."
-      puts "Backtrace: #{e.message}"
-      e.backtrace.each { |line| puts "#{line}\n"}
-      return FailedResponse.new("")
+      elog "No data was returned from the data service for request path : #{path}, message: #{e.message}"
+      return FailedResponse.new('')
```

Comment:
Should we pass in a `nil` for the response objet if we do not have an actual response object? Would there be value in adding an exception parameter to `ResponseWrapper` to encapsulate the exception for cases like this one?

Reply:
as a rule of thumb when choosing between an empty string and null I prefer the empty string (npe prevention).  I don't think so, elog already prints out everything I need